app/src/main/java/protect/card_locker/LoyaltyCard.java

```diff
@@ -97,7 +156,103 @@ public void writeToParcel(Parcel parcel, int i) {
         parcel.writeInt(archiveStatus);
     }
 
-    public static LoyaltyCard toLoyaltyCard(Cursor cursor) {
+    public static LoyaltyCard fromBundle(Bundle bundle) {
+        // Grab default card
+        LoyaltyCard loyaltyCard = new LoyaltyCard();
+
+        // Update from bundle
+        loyaltyCard.updateFromBundle(bundle);
+
+        // Return updated version
+        return loyaltyCard;
+    }
```

Comment:
Incomplete bundles will cause a card that's partially from bundle data and partially default. There will be no warning when we do so. Is this behaviour we accept? If not, what possible options do we have to block this?

Reply:
Made it explicit if a partial update is allowed with the "requireFull" parameter